en/services/component_information.md

```diff
@@ -0,0 +1,91 @@
+# Component Information Protocol
+
+> **Note** This service is still marked as "work in progress", and should not be relied upon in production setups.
+
+The *Component Information Protocol* is a MAVLink service for requesting information from (and about) MAVLink components. 
+It is intended to provide autopilot- and version- independent feature discovery and configuration, allowing a GCS to configure its UI and/or a device without knowing anything about the connected system.
+
+Information shared using this service may include:
+- What types of component information are supported (by this component).
+- What MAVLink commands are supported (both in missions and in other modes).
+- Parameter metadata for parameters supported by the vehicle.
+- Metadata about events emitted by the system
+- Self-describing configuration UIs (i.e. similar to MAVLink camera configuration files).
+- Translations of other metadata.
+
+The service interface uses the [COMPONENT_INFORMATION](#COMPONENT_INFORMATION) message to deliver information about the location of JSON-format [component information files](#schema_files) (and associated translation files).
+
+Component information is requested using [MAV_CMD_REQUEST_MESSAGE](#MAV_CMD_REQUEST_MESSAGE), specifying the [COMPONENT_INFORMATION](#COMPONENT_INFORMATION) message and [type](#COMP_METADATA_TYPE) of component information required.
+
+> **Note**  The `COMP_METADATA_TYPE_VERSION` type provides information about the data types the component supports, and should therefore be requested first.
+
+The `COMPONENT_INFORMATION` message provides a URL to where a file containing the requested information can be downloaded, along with a uid that can be used to determine if the file has changed since last downloaded (effectively a hash for the file). 
+The URL is either a HTTP or MAVLink FTP address; the second is preferred because the file supplied by a vehicle can always be assumed to be up to date.
+
+In addition, the `COMPONENT_INFORMATION` may include a URL and uid for a translation file.
+If supplied, this contains translation strings for the information file specified in the message, in JSON translation file format (superseding any translations in the information file).
```

Comment:
@DonLakeFlyer The "in JSON translation file format " is copied from the message. There appear to be multiple possible formats. Is there a spec for this I can point to?

Reply:
https://github.com/mavlink/qgroundcontrol/blob/master/translations/README.md

Reply:
Thanks @DonLakeFlyer . I have linked to this but when we have an actual end to end translation in the prototype we should link instead to a schema doc and an example translation file.